src/browser/keyboard.js

```diff
@@ -433,13 +447,19 @@ export function KeyboardAdapter(bus)
     }
 
     /**
-     * @param {KeyboardEvent|Object} e
+     * @param {KeyboardEvent|InputEvent|Object} e
      * @param {boolean} keydown
      */
     function handle_event(e, keydown)
     {
         var code = translate(e);
 
+        if(e.inputType)
+        {
+            handle_inputevent(e.inputType, e.data);
```

Comment:
I'm assuming `e.inputType` is defined iff `e` came from an `input` event? In that case I'd suggest removing this code path and passing `handle_inputevent` to `window.addEventListener` directly.